### PR TITLE
feat: implement new report interface

### PR DESCRIPTION
**Pull request by gabrielcsapo**
* Implement new report interface and updates suggestions and UI as a result of the change.
* Updates graphql json resolution
* Updates UI (partially)

Screen Shot 2022-04-26 at 11 24 26 PM

### CODE REVIEW

**Comment by gabrielcsapo** [comment]
@lbdm44 a start to what we talked about online. Will pair on the graphql and cleanup pieces with you tomorrow.

**Comment by gabrielcsapo** [comment]
Some interesting stats,

initial report size: 5.7mbs
report size after change: 1.5mbs

File: packages/ui/components/SuggestionOverview.tsx [hunk 1]
```diff
@@ -24,22 +27,22 @@ function getCostlyPackages(
...
+      //       packageName: associatedActionPackageName,
+      //       count: 1,
+      //     };
+      //     countMap[associatedActionPackageName] = suggestionCount;
+      //     counts.push(suggestionCount);
+      //   } else {
+      //     countMap[associatedActionPackageName].count++;
+      //   }
+      // }
     });
```
Comment (lbdm44) [inline packages/ui/components/SuggestionOverview.tsx:46]: is this meant to be commented out? Seems like the counts won't work if we dont have some sort of log here, maybe there is something further along in the PR that im missing

File: packages/core/src/report/__tests__/__snapshots__/generate-report.test.ts.snap [hunk 2]
```diff
@@ -1,48 +1,57 @@
...
-    {
-      "breadcrumb": "left-pad",
+Report {
+  "dependencies": {
+    "left-pad@1.3.0": {
       "dependencies": [],
-      "funding": "N/A",
-      "homepage": "N/A",
+      "funding": undefined,
+      "homepage": undefined,
```
[the remark this answers is not in the document]
Comment (gabrielcsapo) [reply packages/core/src/report/__tests__/__snapshots__/generate-report.test.ts.snap:9]: It was more of a purposeful change. It should just not be included if the key is undefined.

[the remark this answers is not in the document]
Comment (lbdm44) [reply packages/core/src/report/__tests__/__snapshots__/generate-report.test.ts.snap:9]: 👍

File: packages/ui/components/SuggestionOverview.tsx (hunk 1 above)
Comment (lbdm44) [reply packages/ui/components/SuggestionOverview.tsx:46]: Cool, sounds good

**Review by lbdm44 — APPROVED** [review]

**Pull request merged**